### PR TITLE
Add tap changer automatons

### PR DESCRIPTION
**Pull request by Lisrte**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**
Add the following models :
- LoadTwoTransformers
- LoadOneTransformerTapChanger
- LoadTwoTransformersTapChangers
- TapChangerAutomaton
- TapChangerBlockingAutomaton

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzContext.java [hunk 1]
```diff
@@ -108,7 +110,20 @@ public <T extends Model> T getDynamicModel(Identifiable<?> equipment, Class<T> c
         if (connectableClass.isInstance(bbm)) {
             return connectableClass.cast(bbm);
         }
-        throw new PowsyblException("The model identified by the static id " + equipment.getId() + " does not match not the expected model (" + connectableClass.getSimpleName() + ")");
+        throw new PowsyblException(String.format(MODEL_ID_EXCEPTION, equipment.getId(), connectableClass.getSimpleName()));
+    }
+
+    public <T extends Model> T getPureDynamicModel(String staticId, Class<T> connectableClass) {
+        BlackBoxModel bbm = dynamicModels.stream()
+                .filter(dm -> staticId.equalsIgnoreCase(dm.getDynamicModelId()))
```
[the remark this answers is not in the document]
Comment (Lisrte) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzContext.java:118]: Use of _equals_ instead

File: dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/automatons/TapChangerBlockingAutomatonGroovyExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        void UMeasurement(String staticId) {
+            Bus bus = network.getBusBreakerView().getBus(staticId)
+            if (bus == null) {
+                throw new DslException("Bus static id unknown: " + staticId)
+            }
+            uMeasurement.add(bus)
+        }
+
+        void UMeasurement(List<String> staticIds) {
```
[the remark this answers is not in the document]
Comment (Lisrte) [reply dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/automatons/TapChangerBlockingAutomatonGroovyExtension.groovy:87]: The method name has to match the one used in the groovy script

File: dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/automatons/TapChangerBlockingAutomatonGroovyExtension.groovy [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        List<Load> loads = []
+        List<TwoWindingsTransformer> transformers = []
+        List<Bus> uMeasurement = []
+        List<String> tapChangerAutomatonIds = []
+
+        TCBAutomatonBuilder(Network network) {
+            this.network = network
+        }
+
+        void transformers(String[] staticIds) {
```
[the remark this answers is not in the document]
Comment (Lisrte) [reply dynawaltz-dsl/src/main/groovy/com/powsybl/dynawaltz/dsl/automatons/TapChangerBlockingAutomatonGroovyExtension.groovy:53]: `UMeasurement(String staticId)` is useless, refactor `UMeasurement` methods to match `transformers` processing

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/loads/LoadTwoTransformers.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Laurent Issertial <laurent.issertial at rte-france.com>
+ */
+public class LoadTwoTransformers extends AbstractLoad implements LoadWithTransformers {
+
+    protected static final List<VarMapping> VAR_MAPPING = Arrays.asList(
+            new VarMapping("transformerT_P1Pu_value", "p"),
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/models/loads/LoadTwoTransformers.java:27]: should we call `TransformerSide::getSideSuffix` when there is `T` or `D` in the the variable names? Or call something like `LoadTwoTransformers.getTransformerVarName(side)`?

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/models/transformers/TapChangerModel.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.powsybl.dynawaltz.models.VarConnection;
+
+import java.util.List;
+
+/**
+ * @author Laurent Issertial <laurent.issertial at rte-france.com>
+ */
+public interface TapChangerModel extends Model {
+
+    String TAP_CHANGER_BLOCKING_BLOCKED_T = "tapChangerBlocking_blockedT";
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/models/transformers/TapChangerModel.java:20]: Same kind of remark, we could use `TransformerSide::getSideSuffix`

**Review by flo-dup — APPROVED** [review]

**Pull request merged**